contrib/fonttools/showttf.c

```diff
@@ -307,8 +311,8 @@ exit(0);
 	length = getlong(ttf);
 	cs = filecheck(util,offset,length);
  printf( "%c%c%c%c checksum=%08x actual=%08x diff=%x offset=%d len=%d\n",
-	     tag>>24, (tag>>16)&0xff, (tag>>8)&0xff, tag&0xff,
-	     checksum, cs, checksum^cs,
+	     (int)((tag>>24)&0xff), (int)((tag>>16)&0xff), (int)((tag>>8)&0xff), (int)(tag&0xff),
```

Comment:
shouldn't those first casts be to chars

Reply:
What compiler were you using to test this? I find it a bit dubious that gcc wouldn't accept chars for `%c`.

Reply:
gcc --version
gcc (Mageia 5.5.0-1.mga6) 5.5.0

```bash
$ cp -pR ff ff_test
$ cd ff_test
[ff_test]$ ./bootstrap; ./configure --enable-fontforge-extras --enable-gcc-warnings; make
```
Seems a bit spotty on what it warns.
I removed the (int) around that line and nothing warned this time except for unsigned ints
I went elsewhere and removed it from another line, now I have a bunch...
```code
    6629: printf("\t  Justification priority data for '%c%c%c%c' script, '%c%c%c%c' lang.\n",
	    (stag>>24)&0xff, (stag>>16)&0xff,
	    (stag>>8)&0xff, stag&0xff,
	    (ltag>>24)&0xff, (ltag>>16)&0xff,
	    (ltag>>8)&0xff, ltag&0xff );
```
```bash
showttf.c: In function ‘readttfjustlangsys’:
showttf.c:6629:12: warning: format ‘%c’ expects argument of type ‘int’, but argument 2 has type ‘uint32 {aka unsigned int}’ [-Wformat=]
     printf("\t  Justification priority data for '%c%c%c%c' script, '%c%c%c%c' lang.\n",
            ^
showttf.c:6629:12: warning: format ‘%c’ expects argument of type ‘int’, but argument 3 has type ‘uint32 {aka unsigned int}’ [-Wformat=]
showttf.c:6629:12: warning: format ‘%c’ expects argument of type ‘int’, but argument 4 has type ‘uint32 {aka unsigned int}’ [-Wformat=]
showttf.c:6629:12: warning: format ‘%c’ expects argument of type ‘int’, but argument 5 has type ‘uint32 {aka unsigned int}’ [-Wformat=]
showttf.c:6629:12: warning: format ‘%c’ expects argument of type ‘int’, but argument 6 has type ‘uint32 {aka unsigned int}’ [-Wformat=]
showttf.c:6629:12: warning: format ‘%c’ expects argument of type ‘int’, but argument 7 has type ‘uint32 {aka unsigned int}’ [-Wformat=]
showttf.c:6629:12: warning: format ‘%c’ expects argument of type ‘int’, but argument 8 has type ‘uint32 {aka unsigned int}’ [-Wformat=]
showttf.c:6629:12: warning: format ‘%c’ expects argument of type ‘int’, but argument 9 has type ‘uint32 {aka unsigned int}’ [-Wformat=]
showttf.c: In function ‘readttfjstf’:
```

Reply:
char is probably the same as int on your system. Since the specifier is %c, then if needed, you should be casting to char. Try that and see if the warning goes away.

Reply:
looks like (char) is fine.
gcc needs a bug fixed.
6691 already has (char)
There's a lot of them that probably could go in a function.
Main task was to get it plus other programs to compile - and it/they does now.